src/compiler/typescript_to_rust/azle_generate/src/generators/cross_canister_call_functions/mod.rs

```diff
@@ -38,126 +41,128 @@ struct CrossCanisterCallFunctionNames {
             })
             .collect();
 
-    let notify_functions: Vec<proc_macro2::TokenStream> = cross_canister_call_functions_infos
-        .iter()
-        .map(|cross_canister_call_functions_info| {
-            cross_canister_call_functions_info.notify.rust.clone()
-        })
-        .collect();
+        let notify_with_payment128_functions: Vec<proc_macro2::TokenStream> =
+            cross_canister_call_functions_infos
+                .iter()
+                .map(|cross_canister_call_functions_info| {
+                    cross_canister_call_functions_info
+                        .notify_with_payment128
+                        .rust
+                        .clone()
+                })
+                .collect();
+
+        quote! {
+            #(#call_functions)*
+            #(#call_with_payment_functions)*
+            #(#call_with_payment128_functions)*
+            #(#notify_functions)*
+            #(#notify_with_payment128_functions)*
+        }
+    }
 
-    let notify_with_payment128_functions: Vec<proc_macro2::TokenStream> =
-        cross_canister_call_functions_infos
-            .iter()
-            .map(|cross_canister_call_functions_info| {
-                cross_canister_call_functions_info
-                    .notify_with_payment128
-                    .rust
-                    .clone()
-            })
-            .collect();
+    fn generate_cross_canister_call_functions_infos(&self) -> Vec<CrossCanisterCallFunctionsInfo> {
+        let canister_type_alias_decls =
+            self.get_type_alias_decls_for_system_structure_type(&SystemStructureType::Canister);
 
-    quote! {
-        #(#call_functions)*
-        #(#call_with_payment_functions)*
-        #(#call_with_payment128_functions)*
-        #(#notify_functions)*
-        #(#notify_with_payment128_functions)*
+        canister_type_alias_decls.generate_cross_canister_call_functions_infos()
     }
 }
 
-pub fn generate_cross_canister_call_functions_infos(
-    programs: &Vec<Program>,
-) -> Vec<CrossCanisterCallFunctionsInfo> {
-    let canister_type_alias_decls =
-        programs.get_type_alias_decls_for_system_structure_type(&SystemStructureType::Canister);
-
-    let cross_canister_call_functions_infos =
-        generate_cross_canister_call_functions_infos_from_canister_type_alias_decls(
-            &canister_type_alias_decls,
-        );
-
-    cross_canister_call_functions_infos
+trait GenerateCrossCanisterCallFunctionsInfos {
+    fn generate_cross_canister_call_functions_infos(&self) -> Vec<CrossCanisterCallFunctionsInfo>;
 }
 
-fn generate_cross_canister_call_functions_infos_from_canister_type_alias_decls(
-    canister_type_alias_decls: &Vec<TsTypeAliasDecl>,
-) -> Vec<CrossCanisterCallFunctionsInfo> {
-    canister_type_alias_decls
-        .iter()
-        .fold(vec![], |acc, canister_type_alias_decl| {
+impl GenerateCrossCanisterCallFunctionsInfos for Vec<AzleTypeAlias<'_>> {
```

Comment:
Make sure that all of the names of things that used to be TsTypeAliasDecls are updated to reflect becoming AzleTypeAliases. Make sure that TsTypeAliasDecls are all wrapped in AzleTypeAliases by making sure that they are only directly used in the AzleTypeAlias module